cdisc_rules_engine/utilities/rule_processor.py

```diff
@@ -338,6 +338,8 @@ def is_relationship_dataset(self, domain: str) -> bool:
             result = True
```

Comment:
I had a quick look it seems this function is more to check if a domain can be "related" to another one. That's why CO is listed here even though it's not of class Relationship. At first I thought any domain that contain either RDOMAIN or RSUBJID could be a candidate but it would need to further investigated. I suggest we create another PR solely dedicated to this topic as I don't want to get myself in a rabbit hole.